### PR TITLE
Quick option to set a commonly used date range in to and from field.

### PR DESCRIPTION
**Pull request by RohanAntony**
Test
- Go to Logs page and click on the timer icon. 
- Check each dropdown value to verify if the From and To is being selected as well as dates in logs output is within range

Added an option in filter that looks like a calendar with event marked on it.
image

When selecting this week
image

### CODE REVIEW

**Comment by LucasBrazi06** [comment]
@RohanAntony Quick feedback.
I would have put the selection after the date filters and choose an icon like a stop watch to not confuse it with the ones in the filters.
Thanks!

**Comment by RohanAntony** [comment]
> @RohanAntony Quick feedback. I would have put the selection after the date filters and choose an icon like a stop watch to not confuse it with the ones in the filters. Thanks!

image

**Comment by LucasBrazi06** [comment]
> image

In white theme :-) and is there a way to display it in a button?

**Comment by RohanAntony** [comment]
> > image
> 
> In white theme :-) and is there a way to display it in a button?

image

File: src/app/shared/table/table.component.ts [hunk 1]
```diff
@@ -128,8 +129,30 @@ export class TableComponent implements OnInit, AfterViewInit, OnDestroy {
     }
   }
 
-  public filterChanged(filterDef: TableFilterDef) {
-    this.dataSource.filterChanged(filterDef);
+  public filterChanged(filterDef: TableFilterDef, key?: string) {
+    if(filterDef.id === 'date_menu_dropdown') {
```
Comment (LucasBrazi06) [inline src/app/shared/table/table.component.ts:133]: This code should not be there.
The two date filters should be passed to the DateMenuDropdownFilter object which will update the Json date objects according the drop down selection.
The this code should be in the DateMenuDropdownFilter.
I don't know if the framework is supporting that currently.
You can check with Mahdi who did the dependencies between filters.

File: src/app/shared/table/table.component.html [hunk 2]
```diff
@@ -92,6 +92,21 @@
         <ng-template ngFor let-filterDef [ngForOf]="dataSource.tableFiltersDef">
           <div *ngIf="filterDef.visible !== false"
             [class]="(filterDef.class ? filterDef.class : '')">
+            <!-- Button based Dropdown filter -->
+            <div *ngIf="filterDef.type === FilterType.MENU_DROPDOWN">
+              <button mat-stroked-button [matMenuTriggerFor]="menu">
```
Comment (LucasBrazi06) [inline src/app/shared/table/table.component.html:97]: Use the primary (blue) button design (same as the ones in the toolbar)

Comment (LucasBrazi06) [inline src/app/shared/table/table.component.html:97]: If it's possible also, reduce the width of the button so we have a square button and not a rectangular one.

**Review by LucasBrazi06 — CHANGES_REQUESTED** [review]
Check my comments

**Comment by RohanAntony** [comment]
Closing PR as [Date Range Picker PR](https://github.com/sap-labs-france/ev-dashboard/pull/2895) replaces this.

**Pull request closed without merging**